data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/StreamCheckpointManager.java

```diff
@@ -107,25 +135,52 @@ public void stop() {
         changeEventStatuses.add(changeEventStatus);
         return changeEventStatus;
     }
 
-    public AcknowledgementSet createAcknowledgmentSet(BinlogCoordinate binlogCoordinate) {
+    public ChangeEventStatus saveChangeEventsStatus(LogSequenceNumber logSequenceNumber, long recordCount) {
+        final ChangeEventStatus changeEventStatus = new ChangeEventStatus(logSequenceNumber, Instant.now().toEpochMilli(), recordCount);
+        changeEventStatuses.add(changeEventStatus);
+        return changeEventStatus;
+    }
+
+    public AcknowledgementSet createAcknowledgmentSet(BinlogCoordinate binlogCoordinate, long recordCount) {
         LOG.debug("Create acknowledgment set for events receive prior to {}", binlogCoordinate);
-        final ChangeEventStatus changeEventStatus = new ChangeEventStatus(binlogCoordinate, Instant.now().toEpochMilli());
+        final ChangeEventStatus changeEventStatus = new ChangeEventStatus(binlogCoordinate, Instant.now().toEpochMilli(), recordCount);
         changeEventStatuses.add(changeEventStatus);
+        return getAcknowledgementSet(changeEventStatus);
+    }
+
+    public AcknowledgementSet createAcknowledgmentSet(LogSequenceNumber logSequenceNumber, long recordCount) {
+        LOG.debug("Create acknowledgment set for events receive prior to {}", logSequenceNumber);
+        final ChangeEventStatus changeEventStatus = new ChangeEventStatus(logSequenceNumber, Instant.now().toEpochMilli(), recordCount);
+        changeEventStatuses.add(changeEventStatus);
+        return getAcknowledgementSet(changeEventStatus);
+    }
+
+    private AcknowledgementSet getAcknowledgementSet(ChangeEventStatus changeEventStatus) {
         return acknowledgementSetManager.create((result) -> {
-           if (result) {
-               changeEventStatus.setAcknowledgmentStatus(ChangeEventStatus.AcknowledgmentStatus.POSITIVE_ACK);
-           } else {
-               changeEventStatus.setAcknowledgmentStatus(ChangeEventStatus.AcknowledgmentStatus.NEGATIVE_ACK);
-           }
+            if (result) {
+                positiveAcknowledgementSets.increment();
+                changeEventStatus.setAcknowledgmentStatus(ChangeEventStatus.AcknowledgmentStatus.POSITIVE_ACK);
+            } else {
+                negativeAcknowledgementSets.increment();
+                changeEventStatus.setAcknowledgmentStatus(ChangeEventStatus.AcknowledgmentStatus.NEGATIVE_ACK);
+            }
         }, acknowledgmentTimeout);
     }
 
+    private void checkpoint(final EngineType engineType, final ChangeEventStatus changeEventStatus) {
+        LOG.debug("Checkpoint at {} with record count {}. ", engineType == EngineType.MYSQL ?
+                changeEventStatus.getBinlogCoordinate() : changeEventStatus.getLogSequenceNumber(),
+                changeEventStatus.getRecordCount());
+        streamCheckpointer.checkpoint(engineType, changeEventStatus);
+        recordsCheckpointed.increment(changeEventStatus.getRecordCount());
```

Comment:
`getRecordCount` will be total events processed so far. You may want to just increment it by 1 to make sure the checkpoint thread is working as expected.

Reply:
Updated the metric to increment by 1 at a time.